### PR TITLE
fix(ui): replace new edit icon and glossary feedback

### PR DESCRIPTION
**Pull request by karanh37**
### Describe your changes:

Permission related issues in Glossary
Updating edit icon across OM

I worked on ... because ...

Screenshot 2023-04-17 at 5 11 24 PM
Screenshot 2023-04-17 at 5 11 39 PM

#
### Type of change:

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryTermReferences.tsx [hunk 1]
```diff
@@ -140,6 +140,7 @@ const GlossaryTermReferences = ({
                 }}
               />
             )}
+            {!permissions.EditAll && references.length === 0 && <div>--</div>}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryTermReferences.tsx:143]: same

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryTermSynonyms.tsx [hunk 2]
```diff
@@ -57,6 +57,7 @@ const GlossaryTermSynonyms = ({
           }}
         />
       )}
+      {!permissions.EditAll && synonyms.length === 0 && <div>--</div>}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryTermSynonyms.tsx:60]: same

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/RelatedTerms.tsx [hunk 3]
```diff
@@ -178,6 +178,8 @@ const RelatedTerms = ({
               }}
             />
           ))}
+
+          {!permissions.EditAll && selectedOption.length === 0 && <div>--</div>}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/RelatedTerms.tsx:182]: same

File: openmetadata-ui/src/main/resources/ui/src/components/TagsInput/TagsInput.component.tsx [hunk 4]
```diff
@@ -119,6 +119,7 @@ const TagsInput: React.FC<Props> = ({ tags = [], editable, onTagsUpdate }) => {
         onCancel={() => setIsEditTags(false)}
         onSelectionChange={handleTagSelection}
       />
+      {!editable && tags.length === 0 && <div>--</div>}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TagsInput/TagsInput.component.tsx:122]: same

**Review by chirag-madlani — COMMENTED** [review]
@karanh37 can you verify cursor on all the edit icon and confirm here?

**Review by harshach — APPROVED** [review]

**Comment by harshach** [comment]
@karanh37 @chirag-madlani can we merge this in?

**Comment by chirag-madlani** [comment]
> @karanh37 @chirag-madlani can we merge this in?

@harshach has some cy failures, updated with main will merge once cy is green

**Pull request merged**